options/rtdl/generic/main.cpp

```diff
@@ -199,20 +207,24 @@ extern "C" void *interpreterMain(uintptr_t *entry_stack) {
 	// Add the dynamic linker, as well as the exectuable to the repository.
 #ifndef MLIBC_STATIC_BUILD
 	auto ldso_soname = reinterpret_cast<const char *>(ldso_base + strtab_offset + soname_str);
-	initialRepository->injectObjectFromDts(ldso_soname,
+	auto ldso = initialRepository->injectObjectFromDts(ldso_soname,
 		frg::string<MemoryAllocator> { execfn, getAllocator() },
```

Comment:
Can we initialize the path to an empty string? `execfn` is certainly wrong here.